### PR TITLE
[js-api] Update basic and identity tests for exnref

### PR DESCRIPTION
**Pull request by aheejin**
This updates `basic.tentative.any.js` and `identity.tentative.any.js` tests to use the new instructions (`try_table` and `throw_ref`). In addition to converting existing tests to use the new instruction while maintaining the semantics, I added a new test in
`basic.tentative.any.js` that makes use of all four `catch` clause variants to show all catch clauses works well in JS API tests.

These new tests reauire `--js-flags=--experimental-wasm-exnref` argument to chrome, which is not currently supported in WPT out of the box. I've instead confirmed these run with chrome web tests infrastructure (https://chromium.googlesource.com/chromium/src/+/HEAD/docs/testing/web_tests.md#Running-Web-Tests).

### CODE REVIEW

File: test/legacy/exceptions/js-api/basic.tentative.any.js [hunk 1]
```diff
@@ -73,7 +73,7 @@ promise_test(async () => {
 promise_test(async () => {
   const builder = new WasmModuleBuilder();
   const fnIndex = builder.addImport("module", "fn", kSig_v_v);
-  const tagIndex= builder.addTag(kSig_v_r);
+  const tagIndex = builder.addTag(kSig_v_r);
```
Comment (aheejin) [inline test/legacy/exceptions/js-api/basic.tentative.any.js:76]: Drive-by fix

File: test/js-api/exception/basic.tentative.any.js [hunk 2]
```diff
@@ -73,12 +73,18 @@ promise_test(async () => {
...
+        kExprBlock, kExternRefCode,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchNoRef, tagIndexExternref, 0,
+            kExprCallFunction, fnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
         kExprReturn,
       kExprEnd,
       kExprRefNull, kExternRefCode,
```
Comment (aheejin) [inline test/js-api/exception/basic.tentative.any.js:90]: This is equivalent to
```wat
block $outer
  block $catch_dest (result externref)
    try_table (catch $tag-externref $catch_dest)
      call $fn
    end
    br $outer
  end
  return
end
ref.null extern
```

File: test/js-api/exception/basic.tentative.any.js [hunk 3]
```diff
@@ -100,10 +106,15 @@ promise_test(async () => {
...
+        kExprBlock, kExnRefCode,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchAllRef, 0,
+            kExprCallFunction, fnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
+        kExprThrowRef,
       kExprEnd,
       kExprRefNull, kExternRefCode,
```
Comment (aheejin) [inline test/js-api/exception/basic.tentative.any.js:119]: This is equivalent to
```wat
block $outer
  block $catch_all_ref_dest (result exnref)
    try_table (catch_all_ref $catch_all_ref_dest)
      call $fn
    end
    br $outer
  end
  throw_ref
end
ref.null extern
```

File: test/js-api/exception/basic.tentative.any.js [hunk 4]
```diff
@@ -118,3 +129,54 @@ promise_test(async () => {
...
+            kExprEnd,
+            kExprBr, 2,
+          kExprEnd,
+          kExprDrop,
+          kExprDrop,
+          kExprBr, 1,
+        kExprEnd,
+        kExprDrop,
+      kExprEnd,
+      kExprI32Const, 0,
```
Comment (aheejin) [inline test/js-api/exception/basic.tentative.any.js:168]: This is equivalent to
```wast
block $outer
  block $catch_all_ref_dest (result exnref)
    block $catch_ref_dest (result i32 exnref)
      block $catch_all_dest
        block $catch_dest (result i32)
          try_table (catch $tag-i32 $catch_dest) (catch_all $catch_all_dest) (catch_ref $tag-i32 $catch_ref_dest) (catch_all_ref $catch_all_ref_dest)
            call $fn
          end
          br $outer
        end
        return
      end
      br $outer
    end
    drop
    drop
    br $outer
  end
  drop
end
i32.const 0
```

File: test/js-api/exception/identity.tentative.any.js [hunk 5]
```diff
@@ -31,55 +34,73 @@ test(() => {
...
+      kExprBlock, kWasmVoid,
+        kExprBlock, sig_ie_v,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchRef, jsTagIndex, 0,
+            kExprCallFunction, throwJSTagExnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
+        kExprThrowRef,
       kExprEnd
```
Comment (aheejin) [inline test/js-api/exception/identity.tentative.any.js:50]: This is equivalent to
```wast
block $outer
  block $catch_ref_dest (result i32 exnref)
    try_table (catch_ref $js_tag $catch_ref_dest)
      call $throw_js_tag_exn
    end
    br $outer
  end
  throw_ref
end
```

File: test/js-api/exception/identity.tentative.any.js [hunk 6]
```diff
@@ -31,55 +34,73 @@ test(() => {
...
+      kExprBlock, kWasmVoid,
+        kExprBlock, sig_ie_v,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchRef, wasmTagIndex, 0,
+            kExprCallFunction, throwWasmTagExnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
+        kExprThrowRef,
       kExprEnd
```
Comment (aheejin) [inline test/js-api/exception/identity.tentative.any.js:68]: This is equivalent to
```wast
block $outer
  block $catch_ref_dest (result i32 exnref)
    try_table (catch_ref $wasm_tag $catch_ref_dest)
      call $throw_wasm_tag_exn
    end
    br $outer
  end
  throw_ref
end
```

File: test/js-api/exception/identity.tentative.any.js [hunk 7]
```diff
@@ -31,55 +34,73 @@ test(() => {
...
+      kExprBlock, kWasmVoid,
+        kExprBlock, kExnRefCode,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchAllRef, 0,
+            kExprCallFunction, throwJSTagExnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
+        kExprThrowRef,
       kExprEnd
```
Comment (aheejin) [inline test/js-api/exception/identity.tentative.any.js:86]: This is equivalent to
```wast
block $outer
  block $catch_all_ref_dest (result exnref)
    try_table (catch_all_ref $js_tag $catch_all_ref_dest)
      call $throw_js_tag_exn
    end
    br $outer
  end
  throw_ref
end
```

File: test/js-api/exception/identity.tentative.any.js [hunk 8]
```diff
@@ -31,55 +34,73 @@ test(() => {
...
+      kExprBlock, kWasmVoid,
+        kExprBlock, kExnRefCode,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchAllRef, 0,
+            kExprCallFunction, throwWasmTagExnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
+        kExprThrowRef,
       kExprEnd
```
Comment (aheejin) [inline test/js-api/exception/identity.tentative.any.js:104]: This is equivalent to
```wast
block $outer
  block $catch_all_ref_dest (result exnref)
    try_table (catch_all_ref $wasm_tag $catch_all_ref_dest)
      call $throw_wasm_tag_exn
    end
    br $outer
  end
  throw_ref
end
```

File: test/js-api/exception/identity.tentative.any.js [hunk 9]
```diff
@@ -89,12 +110,17 @@ test(() => {
...
+          kExprTryTable, kWasmVoid, 1,
+            kCatchNoRef, jsTagIndex, 0,
+            kExprCallFunction, throwJSTagExnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
         kExprReturn,
-      kExprEnd
+      kExprEnd,
+      kExprI32Const, 0
```
Comment (aheejin) [inline test/js-api/exception/identity.tentative.any.js:123]: This is equivalent to
```wast
block $outer
  block $catch_dest (result i32)
    try_table (catch $js_tag $catch_dest)
      call $throw_js_tag_exn
    end
    br $outer
  end
  return
end
i32.const 0
```

File: test/js-api/exception/identity.tentative.any.js [hunk 10]
```diff
@@ -103,9 +129,14 @@ test(() => {
...
+      kExprBlock, kWasmVoid,
+        kExprBlock, kWasmI32,
+          kExprTryTable, kWasmVoid, 1,
+            kCatchNoRef, jsTagIndex, 0,
+            kExprCallFunction, throwJSTagExnIndex,
+          kExprEnd,
+          kExprBr, 1,
+        kExprEnd,
         kExprThrow, jsTagIndex,
       kExprEnd
```
Comment (aheejin) [inline test/js-api/exception/identity.tentative.any.js:141]: This is equivalent to
```wast
block $outer
  block $catch_dest (result i32)
    try_table (catch $js_tag $catch_dest)
      call $throw_js_tag_exn
    end
    br $outer
  end
  throw $js_tag
end
```

**Review by aheejin — COMMENTED** [review]
I've added wasm instruction-equivalent of the programs for readability.

**Review by dschuff — APPROVED** [review]

File: test/js-api/exception/basic.tentative.any.js [hunk 11]
```diff
@@ -118,3 +129,54 @@ promise_test(async () => {
...
+
+  const buffer = builder.toBuffer();
+
+  const fn = () => {
+    throw exn;
+  };
+  const {instance} = await WebAssembly.instantiate(buffer, {
+    module: { fn, tagI32: tagI32 }
+  });
+  result = instance.exports.all_catch_clauses();
```
Comment (Ms2ger) [inline test/js-api/exception/basic.tentative.any.js]:
```suggestion
  const result = instance.exports.all_catch_clauses();
```

**Review by Ms2ger — APPROVED** [review]
Thank you. I have not reviewed the changed wasm code.
